index.js

```diff
@@ -225,11 +203,20 @@ ShareDbMongo.prototype.close = function(callback) {
     if (err && err.code === 5101) return callback();
     if (err) return callback(err);
     self.closed = true;
-    self._mongoClient.close(function(err) {
-      if (err) return callback(err);
-      if (!self._mongoPollClient) return callback();
-      self._mongoPollClient.close(callback);
-    });
+    self._mongoClient
+      .then(function(mongoClient) {
+        return mongoClient.close();
+      })
+      .then(function() {
+        return self._mongoPollClient;
+      })
+      .then(function(mongoPollClient) {
+        return mongoPollClient && mongoPollClient.close();
+      })
+      .then(function() {
+        callback(null);
+      })
+      .catch(callback);
```

Comment:
I remembered why I prefer using the 2-arg form of `then()`, from discussion last week.

It's specifically when doing interop back to callbacks. If the `callback(null)` throws, then the `catch` here will invoke the callback a second time.

Using the 2-arg form of `then` does not result in that issue:
```js
.then(function() {
  callback(null);
}, callback)
```

We should also switch to 2-arg form of `then` in the other places where relevant too.

Reply:
Ew. But fine.

For the record, I've also had a quick test with chained promises in this style (eg `promise.then(cb1, cb).then(cb2, cb)`), because I was worried that  catching an early rejection in a chain would rescue and continue chain execution, but it doesn't work like that so it seems okay.